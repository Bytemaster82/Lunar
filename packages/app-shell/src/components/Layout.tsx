import React from 'react';
import withStyles, { css, WithStylesProps } from '@airbnb/lunar/lib/composers/withStyles';

export type Props = {
  children: NonNullable<React.ReactNode>;
};

export class ShellLayout extends React.Component<Props & WithStylesProps> {
  render() {
    const { children, styles } = this.props;

    return <div {...css(styles.layout)}>{children}</div>;
  }
}

export default withStyles(() => ({
  layout: {
    height: '100vh',
  },
}))(ShellLayout);
