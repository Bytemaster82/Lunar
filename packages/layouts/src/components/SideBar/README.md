A vertical sidebar with menu items.

```jsx
import IconUser from ':icons/general/IconUser';
import IconSettings from ':icons/interface/IconSettings';
import IconChat from ':icons/interface/IconChat';
import SideBar, { Item } from '.';

<SideBar accessibilityLabel="Side menu">
  <Item icon={<IconUser accessibilityLabel="Profile" />} />
  <Item icon={<IconSettings accessibilityLabel="Settings" />} />
  <Item icon={<IconChat accessibilityLabel="Chat" />} />
</SideBar>;
```

With labels and an active state.

```jsx
import IconUser from ':icons/general/IconUser';
import IconSettings from ':icons/interface/IconSettings';
import IconChat from ':icons/interface/IconChat';
import SideBar, { Item } from '.';

<SideBar accessibilityLabel="Side menu">
  <Item icon={<IconUser accessibilityLabel="Profile" />} label="Stats" />
  <Item icon={<IconSettings accessibilityLabel="Settings" />} label="Config" active />
  <Item icon={<IconChat accessibilityLabel="Chat" />} label="Chat" />
</SideBar>;
```
