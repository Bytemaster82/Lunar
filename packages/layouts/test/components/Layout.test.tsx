import React from 'react';
import { shallow } from 'enzyme';
import SideBar from '../../src/components/SideBar';
import Layout from '../../src/components/Layout';

describe('<Layout />', () => {
  it('errors if invalid `sideBar` is passed', () => {
    expect(() => {
      shallow(<Layout sideBar="Sidebar">Child</Layout>).dive();
    }).toThrowErrorMatchingSnapshot();
  });

  it('renders with main content', () => {
    const wrapper = shallow(<Layout>Child</Layout>).dive();

    expect(wrapper.find('main')).toHaveLength(1);
    expect(wrapper.find('main').prop('role')).toBe('main');
    expect(wrapper.find('main').text()).toBe('Child');
  });

  it('renders a sidebar', () => {
    const sidebar = <SideBar accessibilityLabel="Test">Item</SideBar>;
    const wrapper = shallow(<Layout sideBar={sidebar}>Child</Layout>).dive();

    expect(wrapper.find('aside')).toHaveLength(1);
    expect(wrapper.find('aside').contains(sidebar)).toBe(true);
  });

  it('renders a before aside', () => {
    const aside = <div>Before</div>;
    const wrapper = shallow(<Layout before={aside}>Child</Layout>).dive();

    expect(wrapper.find('aside')).toHaveLength(1);
    expect(wrapper.find('aside').contains(aside)).toBe(true);
  });

  it('renders an after aside', () => {
    const aside = <div>After</div>;
    const wrapper = shallow(<Layout after={aside}>Child</Layout>).dive();

    expect(wrapper.find('aside')).toHaveLength(1);
    expect(wrapper.find('aside').contains(aside)).toBe(true);
  });

  it('renders all 3 aside types', () => {
    const sidebar = <SideBar accessibilityLabel="Test">Item</SideBar>;
    const before = <div>Before</div>;
    const after = <div>After</div>;
    const wrapper = shallow(
      <Layout sideBar={sidebar} before={before} after={after}>
        Child
      </Layout>,
    ).dive();
    const all = wrapper.find('aside');

    expect(all).toHaveLength(3);
    expect(all.at(0).contains(sidebar)).toBe(true);
    expect(all.at(1).contains(before)).toBe(true);
    expect(all.at(2).contains(after)).toBe(true);
  });
});
