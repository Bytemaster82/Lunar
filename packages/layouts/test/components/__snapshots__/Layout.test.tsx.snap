// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> errors if invalid \`sideBar\` is passed 1`] = `
"Warning: Failed prop type: Invalid prop \`sideBar\` of type \`string\` supplied to \`Layout\`, expected a single ReactElement.
    in Layout"
`;
